Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/SurfaceMagnetism.cs

```diff
@@ -7,96 +7,253 @@
 namespace Microsoft.MixedReality.Toolkit.Utilities.Solvers
 {
     /// <summary>
-    /// SurfaceMagnetism casts rays to Surfaces in the world align the object to the surface.
+    /// SurfaceMagnetism casts rays to Surfaces in the world and aligns the object to the hit surface.
     /// </summary>
     public class SurfaceMagnetism : Solver
     {
-        private const float MaxDot = 0.97f;
-
-        private enum RaycastDirectionEnum
+        #region Enums
+        public enum RaycastDirectionType
         {
-            CameraFacing,
+            CameraFacing = 0,
             ToObject,
             ToLinkedPosition
         }
 
-        private enum OrientModeEnum
+        public enum OrientionMode
         {
-            None,
+            None = 0,
             Vertical,
```

Comment:
Now that this is public (and could be chosen/set at runtime), could you copy the comments from the orientationMode member (i.e. the ones that say what each mode is) so that the enum values here have it as well? Primarily this would be helpful for the generated docs and also for folks using autocomplete.